kythe/go/extractors/bazel/treeset/treeset.go

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2021 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package treeset provides functions for extracting targets that use bazel treesets.
+package treeset
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"bitbucket.org/creachadair/stringset"
+)
+
+// ListSources returns the source files underneath path. If path is a file, it returns a set with
+// a single element. If path is a directory, it returns a set containing all the files
+// (recursively) under path.
+func ListSources(arg string) (stringset.Set, error) {
+	f, err := os.Open(arg)
```

Comment:
Use kythe/go/platform/vfs.Open (you'll need to plumb Contexts around the package).